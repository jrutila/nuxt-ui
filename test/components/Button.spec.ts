// @vitest-environment nuxt
import { describe, it, expect } from 'vitest'
import { mountSuspended } from 'nuxt-vitest/utils'
import Button from '../../src/runtime/components/elements/Button.vue'

describe.only('Button', () => {
  it.each([
    [ 'basic case', { slots: { default: () => 'hello' } } ],
    [ 'leading icon', { props: { leading: true, icon: 'heroicons-check' }, slots: { default: () => 'leading icon' } } ],
  ])('renders %s correctly', async (_, options) => {
    const component = await mountSuspended(Button, options)
    expect(component.html()).toMatchSnapshot()
  })
})
