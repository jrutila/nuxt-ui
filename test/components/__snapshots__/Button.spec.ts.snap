// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders basic case correctly 1`] = `
"<button type=\\"button\\" class=\\"focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center\\">
  <!--v-if-->hello
  <!--v-if-->
</button>"
`;

exports[`Button > renders leading icon correctly 1`] = `
"<button type=\\"button\\" class=\\"focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center\\"><span class=\\"heroicons-check flex-shrink-0 h-5 w-5\\" aria-hidden=\\"true\\"></span>leading icon
  <!--v-if-->
</button>"
`;
